### DISCUSSION TITLE
Running Github Actions on PRs After Explicit Approval

### DISCUSSION THREAD

**Discussion by bkabrda** [community/community#25100] [Actions]
<p>Hey folks, so I was looking at all the various ways that an Action can be triggered on a PR. I *think* that the feature that I need is missing, but wanted to ask first and see if someone solved it.</p>
<p>Basically I have a project with a CI Pipeline that needs to run integration tests (currently the project doesn’t use Github Actions to execute). These integration tests need access to some secret variables in order to run, because they’re operating an actual account of a public service. I’d like to be able to create a Github Action that would execute on PRs *with secrets*, but only after I manually inspect the PR to make sure it’s not malicious. E.g. I’d comment on the PR with “Github, ok to test” (or just pressed a button on the PR page) and only then would the Github Action run.</p>
<p>There are implementations of this approach in different CI systems [1], [2]. Is something similar to this possible right now?</p>
<p>Thanks!</p>
<p>[1] <a href="https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin" rel="nofollow noopener">https://wiki.jenkins.io/display/JENKINS/GitHub+pull+request+builder+plugin</a></p>
<p>[2] <a href="https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&amp;tabs=yaml#comment-triggers" rel="nofollow noopener">https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&amp;tabs=yaml#comment-triggers</a></p>

**Comment by oldskool** [comment, marked as answer]
<p>You should just be able to attach a workflow to the <a href="https://help.github.com/en/articles/events-that-trigger-workflows#issue-comment-event-issue_comment" rel="nofollow noopener">issue_comment event type</a> (pull requests are internally/by the API also seen as a type of issue object):</p>
<pre><code>on:
  issue_comment:
    types: [created]
</code></pre>
<p>Then in your steps, you can do some verification on the comment content and check wheter you are the author. Although the latter probably has no 100% fool-proof way to be verified, depending on your use-case it may be sufficient to do it like that. So, something like (untested pseudo-code):</p>
<pre><code>steps:
  - name: run-something
    if: github.event.issue.comment.body == "Github, OK to test" &amp;&amp; github.event.issue.comment.author == "bkabrda"
    run: &lt;do your stuff here&gt;
</code></pre>

**Comment by bkabrda** [comment]
<p>Ah, I didn’t realize that PRs were just issues. I’ll try doing this, thank you very much!</p>

**Comment by brachipa** [comment]
<p>This works, but I can’t see it in Pull Request checks, on;y in GitHub actions, and only after the merge to master.</p>
<p>How can I cause it to be shown in GitHub PR checks?</p>

**Reply by Splines** [reply]
It also doesn't work for me, I don't even see it in any GitHub Actions run after [experimenting a bit](https://github.com/MaMpf-HD/mampf/pull/642).

Is there another option to achieve this? My main goal is to avoid all unit tests being run with every single commit I push to a pull request. This is important for us as a whole new Docker image is built (parts of that are cached, but still) and that takes quite a long time and consumes unnecessary resources on GitHub servers one could easily avoid (speaking of climate change).

In the best case, one could mark certain jobs in a workflow file as "_require approval_". With this, a button would appear next to where you see the checks of a PR. You (as someone with a specific role on a repository, e.g. admin/collaborator/contributor etc.) can click this button in order to approve the tests.

To work around this lack of feature so far, we created a custom GitHub environment named testing and specified in the settings that it requires approval by certain people. This works fine, however, with this approach, you will get messages à la "You requested a deployment to testing 5 minutes ago" with every commit you push.

This clutters the PR view. It'd be awesome if GitHub could implement the requested feature with a button. Maybe upvote this thread to signal to the team that you'd also love to see such a feature ;)
